dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java

```diff
@@ -57,13 +56,68 @@ public static FeatureReplacementAction EE8_FEATURES() {
     private final Set<String> addFeatures = new HashSet<>();
 
     public FeatureReplacementAction(String removeFeature, String addFeature) {
-        this(Collections.singleton(removeFeature), Collections.singleton(addFeature));
+        this();
+        addFeature(addFeature);
+        removeFeature(removeFeature);
         forceAddFeatures = true;
     }
 
     public FeatureReplacementAction(Set<String> removeFeatures, Set<String> addFeatures) {
-        this.removeFeatures.addAll(removeFeatures);
+        this();
+        addFeatures(addFeatures);
+        removeFeatures(removeFeatures);
+    }
+
+    public FeatureReplacementAction() {
```

Comment:
If we are going to add this ctor that's not valid unless you call other methods, lets add a check to `setup()` that validates that some features additions/removals have been configured (i.e. blow up if `addFeatures.size() == 0 && removeFeatures.size() == 0`)

Reply:
I don't quite understand the purpose of this ctor.